CMakeLists.txt

```diff
@@ -5,6 +5,12 @@
 
 cmake_minimum_required(VERSION 3.19.0 FATAL_ERROR)
 
+if(${CMAKE_UNIT_TESTING})
+    set(PROJECT_NAME unittests)
+    set(LIB_NAME MbedOS)
+    project(${PROJECT_NAME})
+    add_subdirectory(UNITTESTS)
```

Comment:
shouldn't be here jsut subdirectory and the other 3 variables to be set in the unittests CMakelists?

Reply:
Cmake will issue a warning
```
CMake Warning (dev) in CMakeLists.txt:
  No project() command is present.  The top-level CMakeLists.txt file must
  contain a literal, direct call to the project() command.  Add a line of
  code such as

    project(ProjectName)

  near the top of the file, but after cmake_minimum_required().

  CMake is pretending there is a "project(Project)" command on the first
  line.

```

Reply:
This is similar to tests, unittests should be the top level CMakefile, and MBed OS should be included there. The same as we do the applications. unittests are basically apps.

This should be removed then and main cmakefile